test/extensions/filters/http/cache/http_cache_test.cc

```diff
@@ -10,6 +10,98 @@ namespace Extensions {
 namespace HttpFilters {
 namespace Cache {
 
+class RawByteRangeTest : public testing::Test {};
+
+
+using RawByteRangeDeathTest = RawByteRangeTest;
+
+
+TEST_F(RawByteRangeTest, isSuffix) {
+  auto r = RawByteRange(UINT64_MAX, 4);
+  ASSERT_TRUE(r.isSuffix());
+}
+
+TEST_F(RawByteRangeTest, IsNotSuffix) {
+  auto r = RawByteRange(3, 4);
+  ASSERT_FALSE(r.isSuffix());
+}
+
+TEST_F(RawByteRangeDeathTest, IllegalByteRange) {
```

Comment:
I'm not convinced death tests on these conditions are worth the costs of running and maintaining them. Death tests are very slow.

Reply:
Ah, I didn't realize they were slow.... we can delete them. On that note, why do we die when creating an invalid range request instead of having `isValid()` or something similar?

Reply:
Removing all the death tests reduced run time from 2.3s to 0.8s.

Reply:
It's usually simpler to use objects that are guaranteed to be in a valid state. For one thing, you don't have to sprinkle isValid() checks everywhere you want to use them and figure out what to do if they aren't valid. One of the common ways to do this is to declare invalid input to be a programming error, and to crash on receiving it. The calling code is responsible for not providing bad input.

IMO, that makes sense here; the parsing has to handle all sorts of malformed range headers; misordered bounds are one type of them.

 (Other common approaches include exceptions and factory functions, but the added complexity doesn't seem worthwhile here.)

Reply:
That makes sense in principle, tho doesn't that mean you have to check your values before creating the object whenever there's values derived from external sources? Given RawByteRange objects are constructed from client-created `range` headers, it seems like we're duplicating the validity check work for when we call the constructor with the values pulled from the header cause we need to make sure client provided values can't kill the process.

`parseRange` (in the other PR) essentially serves as a factory function, I suppose.

Reply:
parseRange should ensure validity, and this ctor doesn't need to duplicate that. I agree that it could be considered a factory function.